lib/matplotlib/tests/test_axes.py

```diff
@@ -4117,9 +4117,10 @@ def test_eventplot_orientation(data, orientation):
 @image_comparison(['marker_styles.png'], remove_text=True)
 def test_marker_styles():
     fig, ax = plt.subplots()
-    for y, marker in enumerate(sorted(
-            {*matplotlib.markers.MarkerStyle.markers} - {"none"},
-            key=lambda x: str(type(x))+str(x))):
+    for y, marker in enumerate([
+            "none",
+            *sorted({*matplotlib.markers.MarkerStyle.markers} - {"none"},
+                    key=lambda x: str(type(x))+str(x))]):
```

Comment:
I tried but struggled to work out what's going on here with the change?

Reply:
The test simply draws all markers in a single figure, ordered by `str(type(x))+str(x)` (not all `x` have same type: markers can be str (most of them), but also int or, previously, None, hence the need to convert to str).  A previous PR introduced `MarkerStyle("none")`, i.e. "none" was added into the `MarkerStyle.markers` dict; in order to not change the baseline image, I had to remove it from the list of entries.  With this PR, I additionally removed None from the dict; again, in order to not change the baseline image, I had to put it back (using its synonym "none") into the list of entries (at the position where it was previously, i.e. in from of everything else).

Reply:
I would write this as

```
    markers = sorted(matplotlib.markers.MarkerStyle.markers,
                     key=lambda x: str(type(x))+str(x))
    # Since generation of the test image, None was removed but 'none' was
    # added. By moving 'none' to the front (=former sorted place of None)
    # we can avoid regenerating the test image. This can be removed if the
    # test image has to be regenerated for other reasons.
    markers.remove('none')
    markers = ['none', *markers]

    for y, marker in enumerate(markers):
```

Reply:
sure